static/js/panel.js

```diff
@@ -126,4 +172,69 @@ var Panel = new (class Panel {
     this.content.setAttribute("aria-expanded", !hidden);
     this.icon.classList.toggle("expanded");
   }
+
+  copyText(copy, text) {
+    copy.setAttribute("data-copying", "true");
+    navigator.clipboard.writeText(text)
+      .then(function() {
+        copy.classList.add("copied");
+        setTimeout(function() {
+          if (!copy.hasAttribute("data-copying")) {
+            copy.classList.remove("copied");
+          }
+        }, 1000);
+      })
+      .finally(function() {
+        copy.removeAttribute("data-copying");
+      });
+  }
+
+  copyMarkdown(type) {
+    const { node, getText } = this.markdown[type];
+    if (node.disabled) {
+      return;
+    }
+
+    const copy = node.querySelector(".copy");
+    const url = this.permalinkNode ? this.permalinkNode.href : document.location.href;
+    const text = getText(url);;
+
+    this.copyText(copy, text);
+  }
+
+  formatSelectedLines() {
+    const texts = [];
+    let lastLine = -1;
+    for (const line of [...Highlight.selectedLines].sort((a, b) => a - b)) {
+      if (lastLine !== -1 && lastLine != line - 1) {
+        texts.push("...");
+      }
+
+      const lineElem = document.getElementById(`line-${line}`).querySelector(".source-line");
+      texts.push(lineElem.textContent.replace(/\n/g, ""));
```

Comment:
Maybe just `.trim()` rather than `.replace(..)`?

Reply:
`trim` removes WhiteSpace and LineTerminator, so, that doesn't preserve the leading and trailing whitespace.
https://tc39.es/ecma262/#sec-trimstring

It would be a rare case that there's trailing whitespace, so `trimEnd` might work, but I think `replace` for LineTerminator (that's added [here](https://github.com/mozsearch/mozsearch/blob/e464d4af245a3dbbacef87628ba92ef1f3cb3468/tools/src/format.rs#L568)) is safer.
but I realized `g` flag shouldn't be necessary.